packages/reactivity/src/baseHandlers.ts

```diff
@@ -203,7 +203,7 @@ function deleteProperty(target: object, key: string | symbol): boolean {
   const hadKey = hasOwn(target, key)
   const oldValue = (target as any)[key]
   const result = Reflect.deleteProperty(target, key)
-  if (result && hadKey) {
+  if (result && hadKey && !isArray(target)) {
```

Comment:
Wouldn't this also prevent reactivity from being triggered when using the `delete` operator with an array?

* [SFC Playground - this PR](https://deploy-preview-6037--vue-sfc-playground.netlify.app/#eNp9ksFO6zAQRX9l5E1TKWrV1x1KqweoQrAABOwwi+BO2oBjW/a4VIry74wdUVggdpm5947POO7FuXOzQ0RxJqqgfOsIAlJ0a2nazllP0IPHWlF7wBI+alL7TdOgIhig8baDCYcn0kijrAkEXdjB6pQonhcl/Cth+TJNlh/xopjCag29NAApaTXOtN0VnJ+92dYUk3Iy5dCQg000PM8aiG5bE3I2B7eokTCd+bx4Ya801XxcgvG5IOycZj9XANV+se77DDgM1Zyr3H2NRDz4v9Ktel9JMZ4gRVYBrmx2zUcbN6v5aaooBQWGb1pmDtbwHWYsKZTtXKvR37lEHaQ4G4GTVmttP25yj3zE8quv9qjef+m/hWPqSXHvMaA/MNtJo9rvkEZ583iLR/4+iZ3dRs3uP8QH5IuPiXG0XUSzZewfvkx7nV9Ca3ZPYXMkNOFrqQSanEP2S8Fv4fKP1b9xl7NlzvFPE8Mnj+HWPQ==)

This seems to work fine on `main`:
* [SFC Playground - main](https://play.vuejs.org/#eNp9ksFO6zAQRX9l5E1TKWrV1x1KqweoQrAABOwwi+BO2oBjW/a4VIry74wdUVggdpm5947POO7FuXOzQ0RxJqqgfOsIAlJ0a2nazllP0IPHWlF7wBI+alL7TdOgIhig8baDCYcn0kijrAkEXdjB6pQonhcl/Cth+TJNlh/xopjCag29NAApaTXOtN0VnJ+92dYUk3Iy5dCQg000PM8aiG5bE3I2B7eokTCd+bx4Ya801XxcgvG5IOycZj9XANV+se77DDgM1Zyr3H2NRDz4v9Ktel9JMZ4gRVYBrmx2zUcbN6v5aaooBQWGb1pmDtbwHWYsKZTtXKvR37lEHaQ4G4GTVmttP25yj3zE8quv9qjef+m/hWPqSXHvMaA/MNtJo9rvkEZ583iLR/4+iZ3dRs3uP8QH5IuPiXG0XUSzZewfvkx7nV9Ca3ZPYXMkNOFrqQSanEP2S8Fv4fKP1b9xl7NlzvFPE8Mnj+HWPQ==)

Reply:
the issues were also fixed in https://github.com/vuejs/core/pull/5912
[see](https://deploy-preview-5912--vue-sfc-playground.netlify.app/#__DEV__eNp9Uk1v2zAM/SucLnYAwUGWW+AE+yqw7bANa4Eeqh5chbHdypIgyWkKw/+9lIy4RVH0Ioh8j3yPlAb21dri2CPbsNJL19oAHkNvd0K3nTUuwACPVZANB4eHeFQytEeEEQ7OdJBRbSa00NJoH6DzNWwjM89+olIGro1T+0/Z4kyonEuEqUt+s+Kf+fqW4CSSE8whX2x3MAgNEGuMwkKZOs8SIzbIeDwLr1qJ+YJqRw7DQfW+2UDmn7TMxlmvqfRe4Q9UGJCE8wWce6cWNvVY8RUVvJGbhaKA0OVyWg8thoKAnVVVQIoAyma1G4Y0+ziWS4pS9q4PwWj4IknjYSvYayuC7fbpUi4nGpWUy7kr4yx4cnNo6+LeG02vkzwLJk1nW4Xurw0tuRVsM00TsYoW/vg75YLrkZ/zskH58E7+3p9iTrB/Dj26I7masVC5GsMEX1z+wRPdZ7Az+14R+wPwP9Im++hxon3r4/DuFS+5/ZX+WKvrK39xCqj9eahoNDLHxBeMvtn3D0Z/sbsu1qmOHo2Nz+kl9CU=)

Close this one.